src/puppetlabs/kitchensink/core.clj

```diff
@@ -965,6 +965,24 @@ to be a zipper."
   (let [bytes (.getBytes s "UTF-8")]
     (digest/sha-1 [bytes])))
 
+(defn utf8-string->sha256
+  "Compute a SHA-256 hash for the UTF-8 encoded version of the supplied
+  string"
+  [^String s]
+  {:pre  [(string? s)]
+   :post [(string? %)]}
+  (let [bytes (.getBytes s "UTF-8")]
+    (digest/sha-256 [bytes])))
+
+(defn file->sha256
+  "Compute a SHA-256 hash for the given java.io.File object.
+  Uses an InputStream to read the file, so it doesn't load all the contents into
```

Comment:
Oh, I see. I'm not creating InputStream here because the `digest` implementation does that part if you pass a `java.io.File` to it.